kb_sdk_plus.md

```diff
@@ -1,10 +1,98 @@
+and runnable while we make changes to the forked codebase.
+
+## Guidelines
+
+* Backwards incompatible changes are expected in some respects:
+  * R code generation will be removed
+    * Unused in KBase or anywhere else as far as we know
+  * Perl code generation will almost certainly be removed
+    * Perl apps can continue to be supported with the original SDK, or wrapped with
+      Python / Java
+      * https://github.com/boriel/perlfunc for instance
+* Otherwise `kb_sdk_plus` needs to be backwards compatible with `kb_sdk`
+  * It must work with the current KBase backend services
+  * Python and Java apps must continue to be supported
+    * It might be possible to make backwards incompatible changes that are fixed
+      with a recompile of an app or a simple migration guide.
+      This would need careful thought and implementation.
+* Generally simplify the codebase where possible and document where needed
+      
+## Prior to releasing for general development
+
+* Get the tests to pass locally
+  * Remove R support, probably remove Perl support
+  * Do we still need Javascript support?
+* Remove the git submodules and `submodules_hacks` directory
+* See if the `sdkbase` directory is still needed, remove if not
+* Try to remove the `lib` directory
+* Remove `JAR_DEPS`, `JAR_DEPS_BIN`, and `DEPENDENCIES`
+* Understand the purpose of the code in the `entrypoint` script and simplify if possible
+* Use the dockerized JobRunner callback server for tests
+  * E.g. remove the Java callback server implementation
+* Ensure tests are running against EE2 vs. EE1 (e.g. the `KBaseJobService.spec`
+  file is suspicious)
+* Fix docker version issues (see Dakota's patch: https://github.com/kbase/kb_sdk_patch)
+* Convert the build to Gradle
+  * Remove `build.xml`, `travis.yml` and `.classpath`
+* Update to Java 11 at least. Test with newer Java versions if possible
+* Get the tests running in GHA with coverage, dependabot, and trivy
+  * At least look into running macOS and windows tests in GHA
+    * If not, clearly document manual testing requirements before a release
+    * Do we need to support windows?
+    * Resources:
+      * https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories
+      * https://github.com/orgs/community/discussions/69211#discussioncomment-7197681
+      * https://github.com/actions/runner/issues/904
+      
+* Get `kb_sdk_plus` image build running in GHA
+  * Or we could just call it `kb_sdk` since the original container is on Dockerhub
+* Add tests that run `kb-sdk` from the command line vs. just testing java classes
+  * Many issues crop up here
+* Get the SDK running on docker03
+  * Not sure what the issues are here, need to investigate
+* Figure out how to support Argonne staff, where Docker Desktop is banned
+  * Is the docker engine CLI not enough?
+
+## Post general development release
```

Comment:
Could also mention the issues from the kb_sdk repo here -- e.g. an annoying python bug concerning the location of the auth and base modules.

Reply:
(Investigate if it is possible to move/copy issues to this repo)

Reply:
I linked to the kb_sdk issues page; were you thinking of something more than that?

Reply:
well, possibly something more explicit like "assess and migrate issues from <kb_sdk repo url>" but the link is fine